spec/support/client_registry.rb

```diff
@@ -100,12 +100,20 @@ def new_global_client(name)
     # Provides an authorized mongo client on the default test database for the
     # default test user.
     when 'authorized'
+      client_options = {
+        database: SpecConfig.instance.test_db,
+        user: SpecConfig.instance.test_user.name,
+        password: SpecConfig.instance.test_user.password,
+      }.tap do |opts|
+        # specify that the user was created using the SCRAM-SHA-1 mechanism if
+        # auth is turned on and MongoDB version is greater than 4.0
+        # (SCRAM-SHA-256 was introduced in 4.0)
+        opts[:auth_mech] = :scram if SpecConfig.instance.user && SpecConfig.instance.supports_scram_256?
```

Comment:
This is my proposed fix for the [authentication errors](https://evergreen.mongodb.com/task_log_raw/mongo_ruby_driver_ruby_mid__mongodb_version~4.0_topology~sharded_cluster_auth_and_ssl~auth_and_ssl_ruby~ruby_2.5_os~rhel70_test_patch_a92b967961c216d34c74fdea967adc1fa7a68903_5d922b4057e85a5c6654d8c0_19_09_30_16_20_17/0?type=T) in `connection_pool_populator_spec` and `connection_pool_spec`.

Here is why those errors were occurring:
1. I changed `Mongo::Client` to always set a default `auth_source` if you pass in user credentials (as per the authentication spec). This means that the `authorized` and `root_authorized` clients will always have an auth source set in their options.
2. In certain tests, the client options are passed to the app_metadata object via a [connection handshake](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/connection.rb#L309).
3. The app_metadata object sends [different options](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/app_metadata.rb#L76) to the server if it is given an auth source.
4. This causes the server to respond with a list of viable authentication mechanisms where it wasn't before. This is not a problem unless the server version is 4.0 or greater, where both SCRAM-SHA-1 and SCRAM-SHA-256 are both supported. In this case, the post-handshake logic will [set the default auth mech to SCRAM-SHA-256](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/connection.rb#L337).
5. This would not be a problem.... except mlaunch [does not create users using the SCRAM-SHA-256 auth mechanism](https://github.com/rueckstiess/mtools/blob/develop/mtools/mlaunch/mlaunch.py#L1699). So even though SCRAM-SHA-256 is the default auth mechanism for these servers, trying to authenticate with that mechanism will cause an error.

Specifying that the users are created using the SCRAM-SHA-1 mechanism fixes these errors, and it's more explicit, which I like. I'm also open to other opinions.

Reply:
If mlaunch is changed to create users with `opts = {'mechanisms': ['SCRAM-SHA-1','SCRAM-SHA-256']}`, would that fix everything?

Reply:
Yes, I believe so, but there currently isn't a public API for those options. I'm happy to submit a pull to mlaunch at some point. What do you think is the best path forward now?

Reply:
I have two concerns right now:

1. If a developer creates users without specifying any mechanisms, would our test suite still work? I believe it will because I am under the impression that without specifying mechanisms, a user will be created with both scram1 and scram256 mechanisms, but I could be mistaken.

2. The comment above the code being changed in this hunk no longer matches the code.

Assuming the first concern is a non-issue, adjusting the comment to reflect that we are temporarily forcing scram1 even on 4.2+ servers until mlaunch is fixed I think is sufficient for now.

Reply:
You're correct about your first concern -- The tests will still work because users are created with both scram1 and scram256 mechanisms. I'll fix the comment now.

Reply:
@p-mongo is the new comment clearer?

Reply:
Yes I think this is good.